packages/amplify_datastore/lib/types/DataStoreHubEvents/HubEventElement.dart

```diff
@@ -13,22 +13,48 @@
  * permissions and limitations under the License.
  */
 
+import 'dart:ui' show hashValues;
+
 import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
-import 'package:collection/collection.dart';
 
+part 'HubEventElementWithMetadata.dart';
+
+/// The model associated with a DataStore `outboxMutationEnqueued` or
+/// `outboxMutationProcessed` Hub event.
 class HubEventElement {
-  late Model model;
-
-  HubEventElement(
-      Map<dynamic, dynamic> serializedData, ModelProviderInterface provider) {
-    Map<String, dynamic> serializedElement =
-        Map<String, dynamic>.from(serializedData["element"]);
-    Map<String, dynamic> modelData =
-        Map<String, dynamic>.from(serializedElement["model"]);
-    Map<String, dynamic> serializedModelData =
-        Map<String, dynamic>.from(modelData["serializedData"]);
-    model = provider
-        .getModelTypeByModelName(serializedData['modelName'])
-        .fromJson(serializedModelData);
+  /// The instance of the mutated model.
+  final Model model;
+
+  const HubEventElement(this.model);
+
+  factory HubEventElement.fromMap(
+    Map serializedHubEventElement,
+    ModelProviderInterface provider,
+  ) {
+    var model = _parseModelFromMap(serializedHubEventElement, provider);
+    return HubEventElement(model);
   }
+
+  @override
```

Comment:
We definitely need to start doing this more consistently.

Reply:
Yeah, it makes for easier testing and just trying to leave things better than I found found em. I can revert it.

Reply:
Definetely would be good to add, but in a consistent manner.